src/components/CreatePostInput/index.js

```diff
@@ -186,6 +141,53 @@ const CreatePostInput = ({
     }
   };
 
+  const handleHashtagDetection = (v, positionHashtag) => {
+    const spaceStatus = v.includes(' ', positionHashtag);
+    const detectEnter = v.includes('\n', positionHashtag);
+    const textSearch = v.substring(positionHashtag + 1);
+    setHashtagPosition(positionHashtag);
+    /**
+     * cari posisi kursor dimana
+     * cek apakah posisi sebelum kursor # atau bukan
+     * ambil semua value setelah posisi #
+     */
+    if (!spaceStatus && !detectEnter) {
+      setPositionTopicSearch(positionHashtag);
+      searchTopicDebounced(textSearch);
+      setPositionKeyboard('always');
+      setLastTopicInCursor(textSearch);
+    } else {
+      resetTopicSearch();
+      setPositionKeyboard('never');
+      cancelAllRequest();
+      const removeCharacterAfterSpace = textSearch.split(' ')[0];
+      const hashtagLastCharIndex = positionHashtag + removeCharacterAfterSpace?.length;
+      if (hashtagLastCharIndex === positionEndCursor - 1) {
+        const newTopics = joinTopicIntoTopicList(removeCharacterAfterSpace, topics);
+        setTopics(newTopics);
+        // setHashtags(newTopics)
+      }
+
+      setLastTopicInCursor('');
+    }
+  };
+
+  const handleMentionDetection = (v, positionMention) => {
+    const spaceStatus = v.includes(' ', positionMention);
+    const detectEnter = v.includes('\n', positionMention);
+    const textSearch = v.substring(positionMention + 1);
+    setHashtagPosition(positionMention);
+    if (!spaceStatus && !detectEnter) {
+      setPositionTopicSearch(positionMention);
+      searchUserDebounced(textSearch);
+      setPositionKeyboard('always');
+    } else {
+      resetListUsersForTagging();
+      setPositionKeyboard('never');
+      cancelAllRequest();
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new functions `handleHashtagDetection` and `handleMentionDetection` encapsulate the logic for handling hashtag and mention detection respectively. This makes the code more modular and easier to understand.
```diff
+ const handleHashtagDetection = (v, positionHashtag) => {...};
+ const handleMentionDetection = (v, positionMention) => {...};
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->